src/main/webapp/app/exam/manage/exercise-groups/exercise-groups.component.html

```diff
@@ -154,6 +186,143 @@ <h5 class="font-weight-bold mb-0">{{ exerciseGroup.title }}</h5>
                         <td class="align-middle">
                             {{ exercise.maxScore }}
                         </td>
+                        <!-- Programming exercise specific cells -->
+                        <ng-container *ngIf="exerciseGroupToExerciseTypesDict[exerciseGroup.id].includes(exerciseType.PROGRAMMING)">
+                            <ng-container *ngIf="asProgrammingExercise(exercise) as programmingExercise; else emptyProgrammingCells">
+                                <td class="align-middle">
+                                    {{ programmingExercise.shortName }}
+                                </td>
+                                <td>
+                                    <div>
+                                        <span *ngIf="programmingExercise.templateParticipation && programmingExercise.templateParticipation.repositoryUrl">
+                                            <!--Checks if the programming exercise has a setup with VCS and CI, if this not the case
+                                            the links are disabled--->
+                                            <a
+                                                *ngIf="
+                                                    !noVersionControlAndContinuousIntegrationAvailableCheck(programmingExercise.templateParticipation.repositoryUrl);
+                                                    else noVersionControlAndContinuousIntegrationAvailableTemplate
+                                                "
+                                                href="{{ programmingExercise.templateParticipation.repositoryUrl }}"
+                                                target="_blank"
+                                                >Template</a
+                                            >
+                                        </span>
+                                        <ng-template #noVersionControlAndContinuousIntegrationAvailableTemplate>Template</ng-template>
+                                        <jhi-programming-exercise-instructor-status
+                                            *ngIf="
+                                                programmingExercise.templateParticipation &&
+                                                programmingExercise.templateParticipation.results &&
+                                                programmingExercise.templateParticipation.results.length
```

Comment:
```suggestion
  
                                                programmingExercise?.templateParticipation?.results?.length
```
I think optional chaining should work here